src/compiler/scala/tools/nsc/typechecker/Implicits.scala

```diff
@@ -482,7 +482,10 @@ trait Implicits {
       case AnnotatedType(annots, tp)          => core(tp)
       case ExistentialType(tparams, result)   => core(result).subst(tparams, tparams map (t => core(t.info.bounds.hi)))
       case PolyType(tparams, result)          => core(result).subst(tparams, tparams map (t => core(t.info.bounds.hi)))
-      case TypeRef(pre, sym, args)            => typeRef(pre, sym, args.map(core))
+      case TypeRef(pre, sym, args)            =>
+        val coreArgs = args.mapConserve(core)
+        if (coreArgs eq args) tp
+        else typeRef(pre, sym, coreArgs)
```

Comment:
Would `tp.mapOver(this)` also do the trick?

Reply:
It might if `core` was reworked as a `TypeMap`, but I don't think that would be right here. In particular `mapOver` also transforms the prefix which I don't think we want to do.

Reply:
Mapping over the prefix increases the compile time for the benchmark from 25s to 35s on my machine, so I'd say that's a definite no.